app/sidekiq/form1010cg/submission_job.rb

```diff
@@ -16,11 +16,24 @@ class SubmissionJob
     end
 
     def retry_limits_for_notification
+      return [1, 10] if Flipper.enabled?(:caregiver1010)
+
       [10]
     end
 
     def notify(params)
-      StatsD.increment("#{STATSD_KEY_PREFIX}failed_ten_retries", tags: ["params:#{params}"])
+      unless Flipper.enabled?(:caregiver1010)
+        StatsD.increment("#{STATSD_KEY_PREFIX}failed_ten_retries", tags: ["params:#{params}"])
+        return
+      end
+
+      # Increment on the 2nd run (initial run + 1 retry)
+      StatsD.increment("#{STATSD_KEY_PREFIX}applications_retried") if (params['retry_count']).zero?
+
+      # Increment on the 11th run (intial run + 10 retries)
+      if params['retry_count'] == 10
+        StatsD.increment("#{STATSD_KEY_PREFIX}failed_ten_retries", tags: ["params:#{params}"])
+      end
```

Comment:
I was a little confused on this but I think your logic will work fine. Here's an alternative:

```suggestion
      if Flipper.enabled?(:caregiver1010)
        StatsD.increment("#{STATSD_KEY_PREFIX}applications_retried") if params['retry_count'].zero?

        return unless params['retry_count'] == 10
      end

      StatsD.increment("#{STATSD_KEY_PREFIX}failed_ten_retries", tags: ["params:#{params}"])
```

Reply:
Ultimately a case statement on params['retry_count'] would probably be the easiest to reason about. The flipper logic muddies it a bit.

Reply:
I strongly favor positive conditionals for readability... but we'll wind up with an `unless`/`if !` here one way or another, and IMHO the guard clause pattern with `unless Flipper.enabled?(:caregiver1010)` leaves it clearer.

easier to cut out when removing the flag, too - we won't have the history-thrash of the indents.